src/core/components/sidepanels/panels/outline/arborist/ListTree.tsx

```diff
@@ -1,49 +1,138 @@
 import { Tree } from "react-arborist";
 import { cn } from "../../../../../functions/Functions.ts";
-import { useSelectedBlockIds, useSelectedStylingBlocks } from "../../../../../hooks";
-import { ScrollArea } from "../../../../../../ui";
-
-const data = [
-  { id: "1", name: "Unread" },
-  { id: "2", name: "Threads" },
-  {
-    id: "3",
-    name: "Chat Rooms",
-    children: [
-      { id: "c1", name: "General" },
-      { id: "c2", name: "Random" },
-      { id: "c3", name: "Open Source Projects" },
-    ],
-  },
-  {
-    id: "4",
-    name: "Direct Messages",
-    children: [
-      { id: "d1", name: "Alice" },
-      { id: "d2", name: "Bob" },
-      { id: "d3", name: "Charlie" },
-    ],
-  },
-];
+import {
+  useBuilderProp,
+  useSelectedBlockIds,
+  useSelectedStylingBlocks,
+  useHighlightBlockId,
+  useBlocksStore,
+} from "../../../../../hooks";
+import { ScrollArea, Tooltip, TooltipContent, TooltipTrigger } from "../../../../../../ui";
+import { useCallback, useMemo } from "react";
+
+import { getBlocksTree } from "../../../../../functions/Blocks.ts";
+import { TypeIcon } from "../TypeIcon.tsx";
+import { TriangleRightIcon } from "@radix-ui/react-icons";
+
+const removeDuplicates = (nodes, seen = new Set()) => {
+  return nodes.reduce((acc, node) => {
+    if (!seen.has(node._id)) {
+      seen.add(node._id);
+      const children = node.children ? removeDuplicates(node.children, seen) : [];
+      acc.push({ ...node, children });
+    }
+    return acc;
+  }, []);
+};
+
+function debounce(func, delay) {
```

Comment:
use `import { debounce } from "lodash-es"` instead of writing own implementation

Reply:
Noted. I will use the debounce from 'lodash-es'